### PR TITLE
Implement CI for UI, Logical Backup Images to GHCR

### PR DESCRIPTION
**Pull request by dgiebert**
Closes https://github.com/zalando/postgres-operator/issues/2303 https://github.com/zalando/postgres-operator/issues/2313

### CODE REVIEW

**Comment by FxKu** [comment]
@dgiebert can you rebase? I merged #2426 which only provided changes to UI image. Could you split up the steps for all images like it was done in the other PR that I merged? Thanks :)

**Comment by lyz-code** [comment]
@dgiebert would it be possible if you add also support for the pooler? it's the piece that is preventing us from having full ARM support

**Comment by FxKu** [comment]
> I can surely do that! I did split the container image building, but do you also want me to split the name generation then?
Splitting it up sounds cleaner to me.

> Regarding the merged PR #2426 I had to build the frontend and bundle it in, do you prefer to have this done via Github Actions or Multistage?
You mean different Github actions for different tools vs. all tools in multiple stages on one action? If yes, I would prefer the latter.

The pooler file is not part of this OS repo. I will take care of it to make it arm-ready for the release.

**Comment by lyz-code** [comment]
I've been searching for the dockerfile but [it looks like it's not yet open source](https://github.com/zalando/postgres-operator/issues/1964) :( 

@FxKu is it in your roadmap to either opensource the image or to give ARM support? This is preventing us from deploying postgres in HA

**Comment by FxKu** [comment]
@lyz-code seems you did not read the last sentence of my last post. Pooler will get arm support, too.

File: .github/workflows/publish_ghcr_image.yaml [hunk 1]
```diff
@@ -52,23 +59,30 @@ jobs:
...
+      - name: Build and push multiarch ui image to ghcr
         uses: docker/build-push-action@v3
         with:
-          context: ${{ github.workspace }}/ui
-          file: ${{ github.workspace }}/ui/Dockerfile
+          context: ui
           push: true
           build-args: BASE_IMAGE=alpine:3.15
-          tags: "${{ steps.image_ui.outputs.NAME }}"
+          tags: "${{ steps.image.outputs.UI_IMAGE }}"
```
Comment (FxKu) [inline .github/workflows/publish_ghcr_image.yaml:78]:
```suggestion
          tags: "${{ steps.image_ui.outputs.UI_IMAGE }}"
```

File: .github/workflows/publish_ghcr_image.yaml [hunk 2]
```diff
@@ -52,23 +59,30 @@ jobs:
...
+          tags: "${{ steps.image_ui.outputs.UI_IMAGE }}"
           platforms: linux/amd64,linux/arm64
 
+      - name: Build and push multiarch logical-backup image to ghcr
+        uses: docker/build-push-action@v3
+        with:
+          context: docker/logical-backup
+          push: true
+          build-args: BASE_IMAGE=ubuntu:22.04
+          tags: "${{ steps.image.outputs.BACKUP_IMAGE }}"
```
Comment (FxKu) [inline .github/workflows/publish_ghcr_image.yaml:87]:
```suggestion
          tags: "${{ steps.image_lb.outputs.BACKUP_IMAGE }}"
```

**Comment by FxKu** [comment]
I have resolved the conflicts. We do have the pooler image arm-ready to (master-33) but I will bump this in another PR

**Comment by dgiebert** [comment]
Thank you for fixing it; I was busy last week.

Is there still anything I need to do so this gets merged?

> You mean different Github actions for different tools vs. all tools in multiple stages on one action? If yes, I would prefer the latter.

So I adapted the `ui/Dockerfile` to be multistage, but this could also be done in CI instead (I preferred the multi-stage)

**Comment by hughcapet** [comment]
👍

**Comment by FxKu** [comment]
:+1:

**Comment by FxKu** [comment]
Thanks @dgiebert . I will do some further testing with this new UI image.

**Comment by cyrinux** [comment]
Can you please give me please the new docker images path for both operator and the ui please?

**Comment by mmoscher** [comment]
@cyrinux as you can see in the workflow definition, this GHA workflow will only run on tag. Since no new tag was pushed lately, you won't find any of the new images (ui, logical backup) yet.

**Comment by cyrinux** [comment]
> @cyrinux as you can see in the workflow definition, this GHA workflow will only run on tag. Since no new tag was pushed lately, you won't find any of the new images (ui, logical backup) yet.

Thank @mmoscher, didn't see well!

**Pull request merged**